src/main/java/org/jenkinsci/plugins/saml/SamlSecurityRealm.java

```diff
@@ -255,15 +257,15 @@ public HttpResponse doCommenceLogin(final StaplerRequest request, final StaplerR
         String redirectOnFinish = calculateSafeRedirect(from, referer);
         request.getSession().setAttribute(REFERER_ATTRIBUTE, redirectOnFinish);
 
-        RedirectAction action = new SamlRedirectActionWrapper(getSamlPluginConfig(), request, response).get();
-        if (action.getType() == RedirectType.REDIRECT) {
-            LOG.fine("REDIRECT : " + action.getLocation());
-            return HttpResponses.redirectTo(action.getLocation());
-        } else if (action.getType() == RedirectType.SUCCESS) {
-            LOG.fine("SUCCESS : " + action.getContent());
-            return HttpResponses.literalHtml(action.getContent());
+        RedirectionAction action = new SamlRedirectActionWrapper(getSamlPluginConfig(), request, response).get();
+        if (action instanceof SeeOtherAction) {
```

Comment:
Redirections do not work, if the server returns a 302 code this alternative is not evaluated, I guess is not the correct type for redirections.

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/303

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/302

Reply:
FoundAction must be added

Reply:
https://moz.com/learn/seo/redirection, it must manage 301, 302, and 307 codes